j5int/processfamily
# WIP: Fix some unittests

## Changes

### Changes made to processfamily

  - Don't use ProcThreadAttributeHandleListPopen in Python 3
    - On Python 3.7 redirecting the standard handles with `close_fds=True` works
      on Windows so using `ProcThreadAttributeHandleListPopen` is unnecessary.

### Changes made to the tests

  - Replaced `json.dumps(..., encoding=...)` with `json.dumps(...).encode(...)`
  - Add check `sys.std* is None` in addition to `sys.std*.fileno() < 0`
    - In Python 2 when one uses `pythonw.exe` the standard file handles are set
      to an invalid (negative) file descriptor
    - In Python 3 when one uses `pythonw.exe` an the standard handles are set to
      `None` instead of an invalid handle.
  - Refactored unit tests so that instead of inheriting from a base class they
    use pytest fixtures
    - This makes it much easier to select which tests with which fixture you
      want to run on PyCharm and the command line.
  - Disable ExeBuilder in Python 3 (because it doesn't work yet)
  -  Make FunkyWebserver._open_file_handle inheritable.
     - The tests expect file descriptors to be inheritable, but from Python 3.4
       onwards, file descriptors are set to be not inheritable by default.
  - Skip the WindowsServiceTests if you are not running them as Administrator


### Other changes

  - Replaced the deprecated `isAlive` function with `is_alive`
  - Add dependencies `future`, `pytest`, `pytest-lazy-fixture` and remove `nose`
    from `setup.py`


### Weird things

  - In order to get the windows service tests running you must move a dll from
    one folder in the site-packages to another.


## Unit tests status

### Passing tests

TBA

### Failing tests

TBA

## TODO

Still some Python 3 test failing.
